cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -699,11 +695,37 @@ private void extractArchive(Path file, Path targetDir, Function<InputStream, Arc
           mkdirs(entryPath.getParent());
           Files.copy(ais, entryPath);
         }
-        if (isTar && !this.context.getSystemInfo().isWindows()) {
+        if (!this.context.getSystemInfo().isWindows()) {
           Set<PosixFilePermission> permissions = PosixFilePermissions.fromString(permissionStr);
           Files.setPosixFilePermissions(entryPath, permissions);
         }
-        entry = ais.getNextEntry();
+        entry = (TarArchiveEntry) ais.getNextEntry();
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to extract " + file + " to " + targetDir, e);
+    }
+  }
+
+  private void extractZipArchive(Path file, Path targetDir) {
+
+    this.context.trace("Unpacking archive {} to {}", file, targetDir);
+    try (FileInputStream fis = new FileInputStream(file.toFile()); ZipInputStream zis = new ZipInputStream(fis);) {
+      ZipEntry entry = zis.getNextEntry();
```

Comment:
The scan of ZIP files is not necassary anymore, because of last updates in the progressbar class. See here: https://github.com/devonfw/IDEasy/pull/567#discussion_r1754581343